### ISSUE TITLE
Bug: does_arn_match has failing case

### ISSUE THREAD

**Issue by kmcquade** [salesforce/policy_sentry#22]
File in question: 
* `test/test_arns.py`

### Issue description

In the arntable, Greengrass resources start with a /. Alter the parse_arn function in /shared/arns.py so that it can handle the various greengrass resource paths.

Check out the SQLite database for the example.

### Example
Snippets:
```
arn:aws:greengrass:${Region}:${Account}:/greengrass/definition/devices/${DeviceDefinitionId}/versions/${VersionId}
/greengrass/groups/${GroupId}/deployments/${DeploymentId}/artifacts/lambda/${ArtifactId}
/greengrass/definition/functions/${FunctionDefinitionId}/versions/${VersionId}
/greengrass/things/${ThingName}/connectivityInfo
```

There are two cases where it fails:

The Greengrass issue, as mentioned above
test_does_arn_match_case_3 (see the unit tests)

**Comment by kmcquade** [comment]
Tried taking a quick look at this.


```python
    def test_does_arn_match_case_greengrass(self):
        # Undocumented case: AWS Greengrass: arn:aws:greengrass:${Region}:${Account}:/greengrass/definition/devices/${DeviceDefinitionId}/versions/${VersionId}
        arn_to_test = "arn:aws:greengrass:${Region}:${Account}:/greengrass/definition/devices/1234567/versions/1"
        arn_in_database = "arn:aws:greengrass:${Region}:${Account}:/greengrass/definition/devices/${DeviceDefinitionId}/versions/${VersionId}"
        self.assertTrue(does_arn_match(arn_to_test, arn_in_database))

        arn_to_test = "arn:aws:greengrass:${Region}:${Account}:/greengrass/definition/functions/1234567/versions/1"
        arn_in_database = "arn:${Partition}:greengrass:${Region}:${Account}:/greengrass/definition/functions/${FunctionDefinitionId}/versions/${VersionId}"
        self.assertTrue(does_arn_match(arn_to_test, arn_in_database))

        # Let's try to match a definition with a function. Should return False - but it's not.
        arn_to_test = "arn:aws:greengrass:${Region}:${Account}:/greengrass/definition/devices/1234567/versions/1"
        arn_in_database = "arn:${Partition}:greengrass:${Region}:${Account}:/greengrass/definition/functions/${FunctionDefinitionId}/versions/${VersionId}"
        self.assertFalse(does_arn_match(arn_to_test, arn_in_database))

```

For whatever reason, the third case is failing. Will pick this up at a later date.

**Comment by kmcquade** [comment]
I actually fixed this in #143
